File: frontend/src/types/salarySetup.ts
export interface SalarySetupListItem {
  id: number;
  userId: number;
  userName: string;
  monthlySalary: number;
  createdAt: string;
  updatedAt?: string | null;
}

export interface CreateSalarySetupRequest {
  userId: number;
  monthlySalary: number;
}

export interface UpdateSalarySetupRequest {
  userId: number;
  monthlySalary: number;
}

File: backend/CMS.Api/Models/SalarySetup.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace CMS.Api.Models;

[Table("SalarySetups")]
public class SalarySetup
{
    [Key]
    public int Id { get; set; }

    public int UserId { get; set; }
    public User User { get; set; } = null!;

    [Required]
    public decimal MonthlySalary { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime? UpdatedAt { get; set; }

    public int? CreatedById { get; set; }
    public User? CreatedBy { get; set; }

    public int? UpdatedById { get; set; }
    public User? UpdatedBy { get; set; }

    public bool IsDeleted { get; set; } = false;
    public DateTime? DeletedOn { get; set; }
    public int? DeletedById { get; set; }
    public User? DeletedBy { get; set; }
}
